Content.Server/Anomaly/AnomalySynchronizerSystem.cs

```diff
@@ -0,0 +1,158 @@
+
+        SubscribeLocalEvent<AnomalyPulseEvent>(OnAnomalyPulse);
+        SubscribeLocalEvent<AnomalySupercriticalEvent>(OnAnomalySupercritical);
+        SubscribeLocalEvent<AnomalyStabilityChangedEvent>(OnAnomalyStabilityChanged);
+    }
+
+    private void OnPowerChanged(EntityUid uid, AnomalySynchronizerComponent component, ref PowerChangedEvent args)
+    {
+        if (args.Powered)
+            return;
+
+        if (!TryComp<AnomalyComponent>(component.ConnectedAnomaly, out var anomaly))
+            return;
+
+        _anomaly.DoAnomalyPulse(component.ConnectedAnomaly.Value, anomaly);
+        DisconnetFromAnomaly(uid, component, anomaly);
+    }
+
+    private void OnInteractHand(EntityUid uid, AnomalySynchronizerComponent component, InteractHandEvent args)
+    {
+        foreach (var entity in _entityLookup.GetEntitiesInRange(uid, 0.15f)) //is the radius of one tile. It must not be set higher, otherwise the anomaly can be moved from tile to tile
+        {
+            if (TryComp<AnomalyComponent>(entity, out var anomaly))
+            {
+                ConnectToAnomaly(uid, component, anomaly);
+                break;
+            }
+        }
+    }
+
+    private void ConnectToAnomaly(EntityUid uid, AnomalySynchronizerComponent component, AnomalyComponent anomaly)
+    {
+        var auid = anomaly.Owner;
+
+        if (component.ConnectedAnomaly == auid)
+            return;
+
+        component.ConnectedAnomaly = auid;
+        //move the anomaly to the center of the synchronizer, for aesthetics.
+        var targetXform = _transform.GetWorldPosition(uid);
+        _transform.SetWorldPosition(auid, targetXform);
+
+        _anomaly.DoAnomalyPulse(component.ConnectedAnomaly.Value, anomaly);
+        _popup.PopupEntity(Loc.GetString("anomaly-sync-connected"), uid, PopupType.Medium);
+        _audio.PlayPvs(component.ConnectedSound, uid);
+    }
+
+    //TO DO: disconnection from the anomaly should also be triggered if the anomaly is far away from the synchronizer.
+    //Currently only bluespace anomaly can do this, but for some reason it is the only one that cannot be connected to the synchronizer.
+    private void DisconnetFromAnomaly(EntityUid uid, AnomalySynchronizerComponent component, AnomalyComponent anomaly)
+    {
+        if (component.ConnectedAnomaly == null)
+            return;
+
+        _anomaly.DoAnomalyPulse(component.ConnectedAnomaly.Value, anomaly); 
+        _popup.PopupEntity(Loc.GetString("anomaly-sync-disconnected"), uid, PopupType.Large);
+        _audio.PlayPvs(component.ConnectedSound, uid);
+
+        component.ConnectedAnomaly = default!;
+    }
+
+    private void OnAnomalyPulse(ref AnomalyPulseEvent args)
+    {
+        var query = EntityQueryEnumerator<AnomalySynchronizerComponent>();
+        while (query.MoveNext(out var ent, out var component))
+        {
+            if (args.Anomaly != component.ConnectedAnomaly)
```

Comment:
once changed move this entire body into a separate helper and just call it with the port to invoke, can reuse for supercrit